python/examples/cache/cache_fn.py

```diff
@@ -0,0 +1,81 @@
+import asyncio
+import random
+import sys
+import time
+import traceback
+from collections.abc import Awaitable, Callable
+from typing import Generic, ParamSpec, TypedDict, TypeVar
+
+from beeai_framework.cache import BaseCache
+from beeai_framework.errors import FrameworkError
+
+P = ParamSpec("P")
+R = TypeVar("R")
+
+
+class TokenResponse(TypedDict):
+    token: str
+    expires_in: float
+
+
+class CacheFn(Generic[P, R]):
```

Comment:
In TypeScript, this class is part of the framework, whereas in this case, it just exists as part of the examples. Why is that?